False positive on AttributeDynamicLookup subscript

**Describe the bug**
Periphery incorrectly reports `subscript(dynamicMember:)` as unused when extending `AttributeDynamicLookup` to support custom `AttributeScope` types.

When defining a custom `AttributeScope` and the corresponding `AttributeDynamicLookup` subscript (as documented by Apple for extending `AttributedString`), periphery reports:

```
warning: Unused function 'subscript(dynamicMember:)'
```

However, removing this subscript causes compilation to fail because `self.myAttribute` cannot be resolved on `AttributeContainer`.

**Reproduction**
```
import Foundation

// Custom attribute key
enum MyAttribute: AttributedStringKey {
    typealias Value = String
    static let name = "MyAttribute"
}

// Custom attribute scope - periphery reports this as unused
extension AttributeScopes {
    struct CustomAttributes: AttributeScope {
        let myAttribute: MyAttribute
    }
}

// Dynamic member lookup subscript - periphery reports this as unused
extension AttributeDynamicLookup {
    subscript<T: AttributedStringKey>(
        dynamicMember keyPath: KeyPath<AttributeScopes.CustomAttributes, T>
    ) -> T {
        self[T.self]
    }
}

// Extension that uses the custom attribute via dynamic member lookup
extension AttributeContainer {
    var customValue: String? {
        get { self.myAttribute }  // This requires the subscript and scope above
        set { self.myAttribute = newValue }
    }
}

// Main entry point
@main
struct Main {
    static func main() {
        var container = AttributeContainer()
        container.customValue = "hello"
        print(container.customValue ?? "nil")
    }
}
```

Output:

```
/private/tmp/periphery-false-positive/Sources/PeripheryFalsePositive/PeripheryFalsePositive.swift:18:5: warning: Unused function 'subscript(dynamicMember:)'
```

**Environment**
Copy the output from running:
```
3.4.0
swift-driver version: 1.127.14.1 Apple Swift version 6.2.3 (swiftlang-6.2.3.3.21 clang-1700.6.3.2)
Target: arm64-apple-macosx26.0
Xcode 26.2
Build version 17C52
```

--- a/Sources/SourceGraph/Mutators/DynamicMemberRetainer.swift
+++ b/Sources/SourceGraph/Mutators/DynamicMemberRetainer.swift
@@ -10,8 +10,11 @@ final class DynamicMemberRetainer: SourceGraphMutator {
     }
 
     func mutate() throws {
+        // Retain all subscript(dynamicMember:) declarations. This signature is specific to
+        // @dynamicMemberLookup and may be declared in extensions of external types where we
+        // cannot verify the attribute exists.
         for decl in graph.declarations(ofKind: .functionSubscript) {
-            if decl.name == "subscript(dynamicMember:)", decl.parent?.attributes.contains(where: { $0.name == "dynamicMemberLookup" }) ?? false {
+            if decl.name == "subscript(dynamicMember:)" {
                 graph.markRetained(decl)
             }
         }

--- a/Tests/Fixtures/Sources/RetentionFixtures/testRetainsDynamicMemberLookupSubscriptInExternalTypeExtension.swift
+++ b/Tests/Fixtures/Sources/RetentionFixtures/testRetainsDynamicMemberLookupSubscriptInExternalTypeExtension.swift
@@ -1,0 +1,21 @@
+import Foundation
+
+// Extension on external @dynamicMemberLookup type (AttributeDynamicLookup)
+public extension AttributeDynamicLookup {
+    subscript<T: AttributedStringKey>(
+        dynamicMember keyPath: KeyPath<AttributeScopes.FixtureAttributes, T>
+    ) -> T {
+        self[T.self]
+    }
+}
+
+public extension AttributeScopes {
+    struct FixtureAttributes: AttributeScope {
+        let myAttribute: MyFixtureAttribute
+    }
+}
+
+public enum MyFixtureAttribute: AttributedStringKey {
+    public typealias Value = String
+    public static let name = "MyFixtureAttribute"
+}

--- a/Tests/PeripheryTests/RetentionTest.swift
+++ b/Tests/PeripheryTests/RetentionTest.swift
@@ -946,6 +946,14 @@ final class RetentionTest: FixtureSourceGraphTestCase {
         }
     }
 
+    func testRetainsDynamicMemberLookupSubscriptInExternalTypeExtension() {
+        analyze(retainPublic: true) {
+            assertReferenced(.extensionEnum("AttributeDynamicLookup")) {
+                self.assertReferenced(.functionSubscript("subscript(dynamicMember:)"))
+            }
+        }
+    }
+
     func testRetainsCodableProperties() {
         analyze(
             retainPublic: true,

--- a/Tests/Shared/DeclarationDescription.swift
+++ b/Tests/Shared/DeclarationDescription.swift
@@ -105,4 +105,8 @@ struct DeclarationDescription: CustomStringConvertible {
     static func extensionClass(_ name: String, line: Int? = nil) -> Self {
         self.init(kind: .extensionClass, name: name, line: line)
     }
+
+    static func extensionEnum(_ name: String, line: Int? = nil) -> Self {
+        self.init(kind: .extensionEnum, name: name, line: line)
+    }
 }

